resolwe/flow/managers/listener/listener.py

```diff
@@ -49,272 +51,22 @@
-                ).get()
-        for field_name, field_value in mapping.items():
-            logger.debug(
-                __(
-                    "Updating field: {} with value: {}",
-                    model._meta.get_field(field_name),
-                    field_value,
-                )
-            )
-            if isinstance(model._meta.get_field(field_name), JSONField):
-                instance_value = getattr(model_instance, field_name)
-                for key, value in field_value.items():
-                    dict_dot(instance_value, key, value)
-            else:
-                setattr(model_instance, field_name, field_value)
-        model_instance.save(update_fields=mapping.keys())
-        return message.respond_ok("OK")
-
-    def handle_get_model_fields_details(
-        self, message: Message[str]
-    ) -> Response[Dict[str, Tuple[str, Optional[str]]]]:
-        """Get the field names and types for the given model.
-
-        The response is a dictionary which maps names to types.
-        """
-        model = apps.get_model("flow", message.message_data)
-        response = {}
-        for field in model._meta.get_fields():
-            related_model = None
-            if field.is_relation:
-                related_model = field.related_model._meta.label.split(".")[1]
-            response[field.name] = (
-                field.get_internal_type(),
-                not field.null,
-                related_model,
-            )
-        return message.respond_ok(response)
-
-    def handle_get_relations(self, message: Message[int]) -> Response[List[dict]]:
-        """Get relations for the given collection object."""
-        collection = get_objects_for_user(
-            self.contributor,
-            "view_collection",
-            Collection.objects.filter(id=message.message_data),
-        ).get()
-
-        return message.respond_ok(serialize_collection_relations(collection))
-
-    def handle_get_model_fields(
-        self, message: Message[Tuple[str, int, List[str]]]
-    ) -> Response[Dict[str, Any]]:
-        """Return the value of the given model for the given fields.
+from resolwe.storage.models import AccessLog
+from resolwe.utils import BraceMessage as __
 
-        The received message format is
-        (model_name, primary_key, list_of_fields).
+from resolwe.flow.managers import consumer
+from resolwe.flow.managers.protocol import WorkerProtocol
 
-        In case of JSON field the field name can contain underscores to get
-        only the part of the JSON we are interested in.
-        """
-        model_name, model_pk, field_names = message.message_data
-        assert model_name in self.ALLOWED_MODELS_RW
-        permission_name = f"{model_name.lower()}_view"
-        model = apps.get_model("flow", model_name)
+from .basic_commands_plugin import BasicCommands  # Register basic commands plugin.
+from .python_process_plugin import PythonProcess  # Register Python process plugin.
```

Comment:
Python process is implemented in a different app (`resolwe/process`), so it would be nice to move the plugin and registration there.

Reply:
Could do. However, currently the entire `python.process` module is also copied into executor and special care must be taken not to copy this module with it.